app/controllers/qa_controller.rb

```diff
@@ -61,9 +61,13 @@ def breadcrumbs
   end
   helper_method :breadcrumbs
 
+  def current_facet_config
```

Comment:
Do you need to separate this out?

Can you check if `custom_options` exist for current_facet instead?

Reply:
No, it doesn't unfortunately, current_facet is built using the following function

```python
def current_facet
    current_facet = facets[page - 1]
    {
      "question" => current_facet_config["question"],
      "description" => current_facet_config["description"],
      "hint_title" => current_facet_config["hint_title"],
      "hint_text" => current_facet_config["hint_text"],
      "facet" => current_facet
    }
  end
```

Reply:
Can't "custom_options" be added to current_facet?

e.g.

```
  def current_facet
    current_facet = facets[page - 1]
    current_facet_config = qa_config["pages"][current_facet["key"]]
    {
      "question" => current_facet_config["question"],
      "description" => current_facet_config["description"],
      "hint_title" => current_facet_config["hint_title"],
      "hint_text" => current_facet_config["hint_text"],
      "custom_options" => current_facet_config["custom_options"],
      "facet" => current_facet
    }
  end
```

Reply:
I don't think it's a good idea to add this into the current_facet function as `custom_options` is only available when using `single` question type. The current facet is generic across all types.

Reply:
Or:

```
  def current_facet
    current_facet = facets[page - 1]
    current_facet_config = qa_config["pages"][current_facet["key"]]
     config = {
      "question" => current_facet_config["question"],
      "description" => current_facet_config["description"],
      "hint_title" => current_facet_config["hint_title"],
      "hint_text" => current_facet_config["hint_text"],
      "facet" => current_facet
    }
    config.merge("custom_options" => current_facet_config["custom_options"]) if current_facet_config["custom_options"].present?
  end
  
   config
```